Automattic/wp-calypso
# Gutenberg: Refactor selectors to reflect supported block editor modes

#### Changes proposed in this Pull Request

This PR tries to improve the maintenance of the block editor code by better indicating we support 2 flows:
- Gutenframe on all simple sites and eligible JP/AT sites.
- Calypsoified Gutenberg on ineligible JP/AT sites.

In order to achieve this, we have:
- Renamed `isGutenlypsoEnabled` to `isGutenframeEnabled` and moved here the logic that was checking if we can should use Gutenframe flows (previously on `isGutenbergEnabled`).
- `isCalypsoifyGutenbergEnabled` remains unchanged and indicates when we should uso Calypsoify flows.
- `isGutenbergEnabled` now only checks if one of the 2 modes can be used by just calling `isCalypsoifyGutenbergEnabled` and `isGutenberEnabled`.

#### Testing instructions

Simple sites
* Go to `/posts/:simpleSiteSlug`.
* Click on the "Add New Post" button.
* Make sure Calypso loads the iframed block editor.

Jetpack site using plugin 7.3
* Go to `/posts/:jetpackSiteSlug`.
* Click on the "Add New Post" button.
* Make sure Calypso loads the iframed block editor.

Jetpack site using plugin older than 7.3
* Go to `/posts/:jetpackSiteSlug`.
* Click on the "Add New Post" button.
* Make sure Calypso loads the Calypso classic editor.

Atomic site using Jetpack plugin 7.3
* Go to `/posts/:atomicSiteSlug`.
* Click on the "Add New Post" button.
* Make sure Calypso loads the iframed block editor.

Atomic site using Jetpack plugin older than 7.3
* Go to `/posts/:atomicSiteSlug`.
* Click on the "Add New Post" button.
* Make sure Calypso loads the Calypsoified WP Admin block editor.
